snuba/web/split.py

```diff
@@ -159,14 +159,9 @@ def execute(
             logger.warning(
                 "Mismatch in start date on time splitter.",
                 extra={"ast": str(from_date_ast), "legacy": str(from_date)},
+                exc_info=True,
```

Comment:
This seems due to the fact that the legacy query parsing just looks for the first occurrence of a date condition in the query, while the ast (correctly) looks for the most recent datetime. So the ast is providing the right result.

Logging the stack trace before changing the behavior just to verify this on more than a few queries.